Modules/xASL_wrp_BIDS2Legacy.m

```diff
@@ -1,196 +0,0 @@
-        
-        if isempty(SessionID)
-            SessionID = 'unknown'; % default to have at least some output
-        end
-        
-        % Remove iteration for iVisit = 1 -> iterate visit/session in this "x.modules.bids2legacy.BIDS.subjects" (always 1 session per x.modules.bids2legacy.BIDS.subjects)
-        % ExploreASL uses visit as a number (e.g. _1 _2 _3 etc)
-        if nVisits==1
-            pathLegacy_SubjectVisit = fullfile(x.dir.xASLDerivatives, SubjectID);
-            VisitString = '';
-        else
-			if isempty(iVisit)
-                fprintf('\nEmpty session number, setting session number to 1...\n');
-                pathLegacy_SubjectVisit = fullfile(x.dir.xASLDerivatives, [SubjectID '_1']);
-			else
-				% If Visit name is of a form ses-number then use this number otherwise the ID
-				if regexpi(SessionID, 'ses-\d+')
-					pathLegacy_SubjectVisit = fullfile(x.dir.xASLDerivatives, [SubjectID '_' SessionID(5:end)]);
-				else
-					error(['Not yet supporting session names as strings, but only as numbers for session ', SessionID]);
-				end
-			end
-            VisitString = [' visit ' SessionID];
-        end
-        
-        SubjectVisit = [SubjectID VisitString];
-        
-        % Create subject/session directory (to enable reruns for pre-imported or crashed datasets, we need a subject level here/above!)
-        xASL_adm_CreateDir(pathLegacy_SubjectVisit);
-
-        %% 4. Parse modality
-        % Modalities - the BIDS scantypes
-        ModalitiesUnique = unique(bidsPar.BIDS2LegacyFolderConfiguration(2:end, 2));
-        nModalities = length(ModalitiesUnique);
-        xASL_bids_BIDS2Legacy_ParseModality(x.modules.bids2legacy.BIDS, bidsPar, SubjectVisit, iSubjSess, ModalitiesUnique, nModalities, bOverwrite, pathLegacy_SubjectVisit);
-
-    end
-
-end
-% Print new line after track progress bar
-fprintf('   \n');
-
-
-% Get directories of current subject. The BIDS subject name can be prefixed
-% with sub- and suffixed with _1 _2 _3 etc for visits, in the legacy format
-SubjectDirs = xASL_adm_GetFileList(fullfile(x.dir.xASLDerivatives), ['^(|sub-)' x.SUBJECT '(|_\d*)$'], [],[],true);
-
-
-%% 5. Parse M0 of current subject
-ListASL4D = [];
-if ~isempty(SubjectDirs)
-    for iDir=1:numel(SubjectDirs)
-        currentASL = xASL_adm_GetFileList(SubjectDirs{iDir}, '^ASL4D\.nii$', 'FPListRec');
-        ListASL4D = vertcat(ListASL4D, currentASL);
-    end
-end
-% Parse M0s
-if ~isempty(ListASL4D)
-    for iList=1:numel(ListASL4D)
-        xASL_bids_parseM0(ListASL4D{iList});
-        [~, currentNifti] = fileparts(ListASL4D{iList});
-        fprintf('M0 parsed for subject %s image %s ...\n', SubjectID, currentNifti);
-    end
-else
-    warning(['When parsing M0: no ASL4D.nii found for ' SubjectID 'in ' x.dir.xASLDerivatives '...']);
-end
-
-
-%% 6. Clean up
-xASL_bids_BIDS2Legacy_CleanUp(x.dir.DatasetRoot);
-
-
-end
-
-
-
-%% Clean-Up subfunction
```

Comment:
Why do we delete this function without replacement? Don't we want to merge and save the log files?

Reply:
See below the code that we then effectively remove.

To me it seems this code merges import log files, which is not something we want in processing-BIDS2Legacy, especially not because both could potentially be run at the same time. (converting to BIDS is completely separate from loading BIDS).

The other thing it does, is moving import log files to a derivatives folder. That is something that should only happen in the import right?

So to me it seemed that the deleted code (below), is rudimentary, a remnant of when BIDS2Legacy was part of the import module.

>>
%% Clean-Up subfunction
function xASL_bids_BIDS2Legacy_CleanUp(pathStudy)

    % Start with empty file list
    filesCleanUp = {};
    
    % Search for dcm2niix summary file
    dcm2niixImportLogFile = xASL_adm_GetFileList(pathStudy,'^import_log.+$');
    
    % Merge log file lists
    if ~isempty(dcm2niixImportLogFile)
        filesCleanUp = vertcat(filesCleanUp,dcm2niixImportLogFile);
    end
    
    % Move files to derivatives
    if ~isempty(filesCleanUp)
        for iFile = 1:size(filesCleanUp,1)
            sourceCleanUp = filesCleanUp{iFile};
            [~, fileCleanUp, extCleanUp] = xASL_fileparts(sourceCleanUp);
            destCleanUp = fullfile(pathStudy, 'derivatives', 'ExploreASL', [fileCleanUp extCleanUp]);
            if xASL_exist(sourceCleanUp,'file')
                xASL_Move(sourceCleanUp,destCleanUp);
            end
        end
    end
    
    % Print new line after track progress bar
    fprintf('   \n');

end

Reply:
Sure. We probably don't want to do that in BIDS2Legacy, but in import. But still, we want to take all the import logs and save them to derivatives. Cause we shouldn't save logs to rawdata and we save all logs to derivatives. 

So the question still comes - why it would help to not do this cleanup that saves the dcm2nii import logs to derivatives?

Reply:
You are right. This code didn't do anything and could be deleted. What I couldn't understand was why the BIDS2Legacy output was so short now. 
Which I understand now - the bVerbose is set to false by default. But I'm not sure this is a good idea. 
Line 77 of xASL_module_BIDS2Legacy can indeed be suppressed. But line 178 should be reported. Can you revise the use of bVerbose there?